src/api/app/models/project.rb

```diff
@@ -1122,6 +1122,12 @@ def open_requests_with_by_project_review
                                                        .joins(:reviews), name).call
   end
 
+  def open_unapproved_requests_with_by_project_review
+    # Includes also by_package reviews for packages contained in this project
+    OpenRequestsWithByProjectReviewFinder.new(BsRequest.where(state: [:new, :review]).where.not(approver: nil)
```

Comment:
And so this call will find requests with by_project review and approver - but you're looking for unapproved requests (according to the name of the function) and those do *not* have an approver. 

What you really want to look at is if all staged requests are approved - those are listed in `@staging_project.staged_requests`.

Reply:
Yeah I have some commits not pushed that change/test this. I got sidetracked by some nasty bug in `BsRequest.auto_accept`...

Reply:
we talked offline

![this is fine...](https://media.giphy.com/media/QMHoU66sBXqqLqYvGO/giphy.gif)